js/src/scrollspy.js

```diff
@@ -150,15 +149,7 @@ class ScrollSpy extends BaseComponent {
       ...(typeof config === 'object' && config ? config : {})
     }
 
-    if (typeof config.target !== 'string' && isElement(config.target)) {
```

Comment:
Are you sure this wasn't needed? IIRC it was used to make sure the IDs are unique or something.

Reply:
It was used to make string selectors, But wasn't using the Selector engine scoped search. 
I replaced the search to be done on element scope, so we keep the element as reference, not the id